### ISSUE TITLE
Inconsistent update behavior for SCIM /Users/{userId}

### ISSUE THREAD

**Issue by robl-sap** [cloudfoundry/uaa#2671]
### What version of UAA are you running?

latest 

### How are you deploying the UAA?

not relevant, but docker, cf deployment

### What did you do?

Update the origin of a specific useruser via SCIM API /Users/{userId} , either via PATCH or PUT.

### What did you expect to see? What goal are you trying to achieve with the UAA?

Consistent behavior that both http methods either allow or disallow the update of the origin of a user via scim.

### What did you see instead?

- PUT allows update of the origin of a user
- PATCH does not allow the update of the origin of a user

```sh
# create a new user
curl -X POST '<uaadomain>/Users' -H "Authorization: bearer $TOKEN" -H "Content-type: application/json" -d '{"userName": "user-does-not-exist",  "origin": "does-not-exist", "emails": [{"value": "test@email.com", "primary": false}]}'

# try to update origin via patch, returns successfully but with old origin
curl -X PATCH '<uaadomain>/Users/<userid>' -H "Authorization: bearer $TOKEN" -H "Content-type: application/json" -H 'If-Match: *' -d '{"origin":"still-not-existing"}'

# update origin via put, returns successfully with new origin
curl -X PUT '<uaadomain>/Users/<userid>' -H "Authorization: bearer $TOKEN" -H "Content-type: application/json" -H 'If-Match: *' -d '{"userName": "user-does-not-exist",  "origin": "still-does-not-exist", "emails": [{"value": "test@email.com", "primary": false}], "name": {"familyName":"familyName", "givenName":"givenName"}}'

```

Why ?

PATCH does not take into consideration the origin while applying the provided delta.

https://github.com/cloudfoundry/uaa/blob/4dae99997cb27c3fea97a07ced3a2a2402f2e037/model/src/main/java/org/cloudfoundry/identity/uaa/scim/ScimUser.java#L711

PUT simply takes the provided user from the body and updates

https://github.com/cloudfoundry/uaa/blob/develop/server/src/main/java/org/cloudfoundry/identity/uaa/scim/endpoints/ScimUserEndpoints.java#L262

Current documentation mentions origin as update field for both methods.

Further, this suffers from the same issue as #2576 , i.e., i can PUT a non-existing origin.

**Comment by hsinn0** [comment]
We will plan to prioritize this.
